Sources/Managers/ApiManager.swift

```diff
@@ -18,7 +18,27 @@ class ApiManager {
         }
     }
 
-    func getUrlRequest(searchTerm: String) throws -> URLRequest? {
+    static var decoder: JSONDecoder {
+        let decoder = JSONDecoder()
+        decoder.keyDecodingStrategy = .convertFromSnakeCase
+        decoder.dateDecodingStrategy = .iso8601
+        return decoder
+    }
+
+    static func getUrlRequest(user: String) throws -> URLRequest {
+        var urlComponents = URLComponents()
+        urlComponents.scheme = "https"
+        urlComponents.host = "api.github.com"
+        urlComponents.path = "/users/\(user)/repos"
+
+        guard let url = urlComponents.url else {
+            throw ApiError.invalidURL
+        }
+
+        return URLRequest(url: url)
+    }
+
+    static func getUrlRequest(searchTerm: String) throws -> URLRequest {
```

Comment:
Same naming for two different methods should be avoided, instead you can do powerful stuff with enums:


```swift
class ApiManager {
    static let shared: ApiManager = ApiManager() // Singleton
    
    // function does not need to be static. you can call it anywhere in the app like so:
    // ApiManager.shared.fetch(model: Repository.self, endpoint: .searchRepositories(query: "php"))
    func fetch<T:Codable>(model: T.Type, endpoint: Endpoint) async throws -> T {
        let (data, response) = try await URLSession.shared.data(for: endpoint.request)
        // Check status code
        // Decode stuff etc.
    }
}

enum Endpoint {
    case userRepositories(userId: String)
    case searchRepositories(query: String)
    
    var request: URLRequest {
        var urlComponents = URLComponents()
        urlComponents.scheme = "https"
        urlComponents.host = "api.github.com"
        urlComponents.path = self.path
        
        guard let url = urlComponents.url else {
            fatalError("Whoops, seems like there is some faulty data when trying to create url!!")
        }
        return URLRequest(url: url)
    }
    
    var path: String {
        switch self {
        case .userRepositories(let userId):
            return "/users/\(userId)/repos"
        case .searchRepositories(let query):
            return "/search/repositories?q=\(query)"
        }
    }
}
```